src/test/groovy/org/prebid/server/functional/tests/privacy/AuctionGppSpec.groovy

```diff
@@ -0,0 +1,226 @@
+package org.prebid.server.functional.tests.privacy
+
+import org.prebid.server.functional.model.request.auction.BidRequest
+import org.prebid.server.functional.model.request.auction.Regs
+import org.prebid.server.functional.model.request.auction.User
+import org.prebid.server.functional.model.response.auction.ErrorType
+import org.prebid.server.functional.tests.BaseSpec
+import org.prebid.server.functional.util.PBSUtils
+import org.prebid.server.functional.util.privacy.CcpaConsent
+import org.prebid.server.functional.util.privacy.TcfConsent
+import org.prebid.server.functional.util.privacy.gpp.TcfEuV2Consent
+import org.prebid.server.functional.util.privacy.gpp.UspV1Consent
+
+import static org.prebid.server.functional.model.request.GppSectionId.TCF_EU_V2
+import static org.prebid.server.functional.model.request.GppSectionId.USP_V1
+import static org.prebid.server.functional.util.privacy.CcpaConsent.Signal.ENFORCED
+import static org.prebid.server.functional.util.privacy.CcpaConsent.Signal.NOT_ENFORCED
+import static org.prebid.server.functional.util.privacy.TcfConsent.GENERIC_VENDOR_ID
+import static org.prebid.server.functional.util.privacy.TcfConsent.PurposeId.BASIC_ADS
+
+class AuctionGppSpec extends BaseSpec {
+
+    def "PBS should populate gdpr to 1 when regs.gdpr is not specified and gppSid contains 2"() {
+        given: "Default bid request with gppSid and without gdpr"
+        def gppSidsId = [TCF_EU_V2.valueAsInt]
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs = new Regs(gdpr: null, gppSid: gppSidsId)
+        }
+
+        when: "PBS processes auction request"
+        def bidResponse = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain regs.gdpr and gppSid from request"
+        def resolvedRequest = bidResponse.ext.debug.resolvedRequest.regs
+        assert resolvedRequest.gdpr == 1
+        assert resolvedRequest.gppSid == gppSidsId
+    }
+
+    def "PBS should populate gdpr to 0 when regs.gdpr is not specified and gppSid not contains 2"() {
+        given: "Default bid request with gppSid and without gdpr"
+        def gppSidsId = [PBSUtils.getRandomNumber(TCF_EU_V2.valueAsInt)]
```

Comment:
This looks different. You have a separate method for getting a number with the exception.